### PR TITLE
add deployhostedfunction

### PR DESCRIPTION
**Pull request by noahdietz**
Adds the `deployhostedfunction` command

### CODE REVIEW

**Comment by jbowen93** [comment]
LGTM

File: test/fixtures/hellohostedfunctions/server.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var http = require('http');
+
+console.log('node.js application starting.. 123.');
```
Comment (AdamMagaluk) [inline test/fixtures/hellohostedfunctions/server.js]: Can we remove the `123.`

File: lib/commands/deployhostedfunction.js [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+var ziputils = require('../ziputils');
+var parseDeployments = require('./parsedeployments');
+var fetchproxy = require('./fetchproxy');
+var deployproxy = require('./deployproxy');
+
+var DeploymentDelay = 60;
+var ProxyBase = 'apiproxy';
+
+// By default, do not run NPM remotely
+var DefaultResolveModules = false;
```
Comment (AdamMagaluk) [inline lib/commands/deployhostedfunction.js]: Unused variable.

File: test/fixtures/hellohostedfunctions/server.js (hunk 1 above)
Comment (jbowen93) [reply test/fixtures/hellohostedfunctions/server.js]: ack

File: lib/commands/deployhostedfunction.js (hunk 2 above)
Comment (jbowen93) [reply lib/commands/deployhostedfunction.js]: ack

**Pull request merged**